### ISSUE TITLE
__slots__ cannot be used as a state id

### ISSUE THREAD

**Issue by srkunze** [saltstack/salt#47351]
Based on #46516

When salt renders

```yaml
__slot__:salt:custom.get_user_home(tunnel):
  file.managed:
    - source: salt://tunnel/files/.ssh/authorized_keys
    - makedirs: True
    - user: tunnel
```

The state is not executed. It does not show up in the result overview.

It's just not there. No error message whatsoever.

**Comment by gtmanfred** [comment]
I have verified this.  As a work around, slots do work in the `-name: ` field, so you would need to just put it there.

I am marking this as a bug though.

Thanks for reporting,
Daniel

**Comment by gtmanfred** [comment]
@DmitryKuzmenko when you get a minute, could you look at this?

Thanks!
Daniel

**Comment by srkunze** [comment]
Not sure if it makes sense to allow in the first place because of dynamic execution of `__slots__`.

I could live with a big error screaming at me that `__slots__` cannot be used as state IDs instead of **silently** dropping the state at all.

**Comment by DmitryKuzmenko** [comment]
During compile high data salt skips states starting with `__` symbols. Such states are a kind of special states like `__extend__` or it could be custom macros that could be used with `use` like proposed here: https://github.com/saltstack/salt/issues/3460#issuecomment-12732273
So if we'd like to keep that states we can just allow to handle `__slots__` in `compile_high_data` like this:
```diff
diff --git a/salt/state.py b/salt/state.py
index 6f18e4e4d6..20bf2e59b5 100644
--- a/salt/state.py
+++ b/salt/state.py
@@ -1337,7 +1337,7 @@ class State(object):
         '''
         chunks = []
         for name, body in six.iteritems(high):
-            if name.startswith('__'):
+            if name.startswith('__') and not name.startswith('__slot__:'):
                 continue
             for state, run in six.iteritems(body):
                 funcs = set()
```
In this case using this simple test state:
```yaml
__slot__:salt:test.echo(text=/tmp/file-c):
  file.copy:
    - source: __slot__:salt:test.echo(text=/etc/hosts)
```
I'm getting the following result:
```
alpha:
----------
          ID: __slot__:salt:test.echo(text=/tmp/file-c)
    Function: file.copy
      Result: True
     Comment: Copied "/etc/hosts" to "/tmp/file-c"
     Started: 12:48:08.586811
    Duration: 832.166 ms
     Changes:   
              ----------
              /tmp/file-c:
                  /etc/hosts

Summary for alpha
------------
Succeeded: 1 (changed=1)
Failed:    0
------------
Total states run:     1
Total run time: 832.166 ms
```

So we can handle this issue out I have just a couple questions:
1. Shall we do this?
2. Do we have to also handle slots in the state ID or raw data for ID is OK?

@thatch45 could you please guide me to the right path?

**Comment by mchugh19** [comment]
Not stale. Seems to be waiting on a direction steer from @thatch45

**Comment by twangboy** [comment]
Closing this issue due to age and lack of activity. Please test this on version 3006.2 and create a new issue if the problem persists. The new issue template has more information and will allow us to track and reproduce the issue more effectively. Thanks!
